### ISSUE TITLE
Monorepo Releases Not getting the latest tag/release

### ISSUE THREAD

**Issue by eliranw** [Justintime50/homebrew-releaser#24]
I'm using homebrew-releaser in a monorepo with multiple projects.

When creating a release called "xxx-xxx-<semantic version>", homebrew-releaser is run and does not find the latest release, but a different release for a different project in the repo. Is there a way to send the specific wanted release name to homebrew-releaser?

**Comment by Justintime50** [comment]
Homebrew Releaser currently does not support monorepos, though that's an interesting feature request. There would need to be a decent amount of rework to make it function in this way. If you could provide some additional details on how this would get used or what the config might look like I'd consider adding it.

**Comment by eliranw** [comment]
Basically today I run my workflows as so:

```
on:
  push:
    tags:
      - '<project>-cli-*'
jobs:
  create-release:
    name: Create Git Release
    steps:
    ..... # Create a github release
  releases-matrix:
    needs: [create-release]
    name: Create Go Binaries
    strategy:
      matrix:
        os: [darwin-amd64,darwin-arm64,windows-amd64,linux-amd64]
    steps:
       ....... # Generate release binaries
  homebrew-releaser:
    needs: [releases-matrix]
    name: homebrew-releaser
    steps:
      - name: Release my project to my Homebrew tap
        uses: Justintime50/homebrew-releaser@v1
        with:
           ......
          target_darwin_amd64: true
          target_darwin_arm64: true
          target_linux_amd64: true
```

My repo has 2 types of releases.
1. A CLI release that has multiple binaries that I want to run homebrew-releaser on.
2. A Helm chart release that does not create binaries.

The action works up until the part where it gets the latest tag. The tag it gets from the API is incorrect, and then it proceeds to try to get the binaries from the release of the tag, and fails. This is because the tag is not the current tag that the pipeline is running on, but a different tag for a release that does not include binaries.

For a tag/release named kubetunnel-cli-<version-number>, the log looks like this:
```
2022-11-02 08:52:56,547 - INFO - Starting Homebrew Releaser...
2022-11-02 08:52:56,547 - DEBUG - All required environment variables are present.
2022-11-02 08:52:56,547 - INFO - Setting up git environment...
Cloning into 'homebrew-tap'...
2022-11-02 08:52:56,8[18](https://github.com/we-dcode/kubetunnel/actions/runs/3376077493/jobs/5603432325#step:3:19) - DEBUG - Git environment setup successfully.
2022-11-02 08:52:56,818 - INFO - Collecting data about kubetunnel...
2022-11-02 08:52:56,933 - DEBUG - HTTP GET request made successfully to https://api.github.com/repos/we-dcode/kubetunnel.
2022-11-02 08:52:57,043 - DEBUG - HTTP GET request made successfully to https://api.github.com/repos/we-dcode/kubetunnel/tags.
2022-11-02 08:52:57,043 - INFO - Latest release (kubetunnel-operator-0.0.16) successfully identified! <----*WRONG RELEASE*
2022-11-02 08:52:57,043 - INFO - Generating tar archive checksum(s)...
2022-11-02 08:52:57,356 - DEBUG - HTTP GET request made successfully to https://github.com/we-dcode/kubetunnel/archive/kubetunnel-operator-0.0.16.tar.gz.
2022-11-02 08:52:58,485 - DEBUG - kubetunnel-operator-0.0.16.tar.gz written successfully.
2022-11-02 08:52:58,566 - DEBUG - Checksum for kubetunnel-operator-0.0.16.tar.gz generated successfully: 860c3be37664bcf9bc46c32697[19](https://github.com/we-dcode/kubetunnel/actions/runs/3376077493/jobs/5603432325#step:3:20)3a02a29135be5b4271ae15fbe825a60dbe8a
[20](https://github.com/we-dcode/kubetunnel/actions/runs/3376077493/jobs/5603432325#step:3:21)[22](https://github.com/we-dcode/kubetunnel/actions/runs/3376077493/jobs/5603432325#step:3:23)-11-02 08:52:58,567 - DEBUG - checksum.txt written successfully.
2022-11-02 08:52:58,778 - DEBUG - HTTP GET request made successfully to https://github.com/we-dcode/kubetunnel/archive/kubetunnel-operator-0.0.16.zip.
2022-11-02 08:53:00,095 - DEBUG - kubetunnel-operator-0.0.16.zip written successfully.
2022-11-02 08:53:00,176 - DEBUG - Checksum for kubetunnel-operator-0.0.16.zip generated successfully: b2dac04[31](https://github.com/we-dcode/kubetunnel/actions/runs/3376077493/jobs/5603432325#step:3:32)dad1d6c975f541c4e3d361caa5f9ee7f7a619cc1bddc88b07661688
2022-11-02 08:53:00,176 - DEBUG - checksum.txt written successfully.
404 Client Error: Not Found for url: https://github.com/we-dcode/kubetunnel/releases/download/kubetunnel-operator-0.0.16/kubetunnel-kubetunnel-operator-0.0.16-darwin-amd64.tar.gz
```

My suggested solution is to add the tag/version as a parameter to the action and the API will request binaries from this specific tag. That parameter would be used [here](https://github.com/Justintime50/homebrew-releaser/blob/0f80d319e73a7a3d9f35fe3711439e07f30887b4/homebrew_releaser/app.py#L76)

**Comment by Justintime50** [comment]
Interesting. This is something we could maybe support in the future, though I have no immediate plans to add support by the end of the year.

The suggested feature set would require a decent amount of refactoring how the action works. I'll keep thinking about this over the holidays.

**Comment by Justintime50** [comment]
After looking into supporting monorepos, I've determined it's not something I want to pursue at this time. It would require a complete re-architecture of how this action works. Possible alternatives would be to split your monorepo up and use this action for each individual repo, find another action that supports monorepos, or build a custom solution to release monorepo formulas.

I have clarified in the README that this project is not compatible with monorepos. If you are interested in putting up a proof of concept PR on how we can get this action working with monorepos, I may reconsider; however, due to the changes needed to make that work, I won't be able to pursue it at this time.
